packages/cloud/src/cloud-operations-client.ts

```diff
@@ -0,0 +1,552 @@
+  }
+
+  /**
+   * Raw gRPC access to the Temporal Cloud Operations service.
+   *
+   * **NOTE**: The API Version provided in {@link options} is **not** automatically set on requests
+   * made via the raw gRPC service object. If the namespace requires it, you may need to do the following:
+   *
+   * ```
+   * const metadata: Metadata = { ['temporal-cloud-api-version']: apiVersion }
+   * const response = await client.withMetadata(metadata, async () => {
+   *   return client.cloudService.getNamespace({ namespace });
+   * });
+   * ```
+   */
+  get cloudService(): CloudService {
+    return this.connection.cloudService;
+  }
+}
+
+/**
+ * @experimental
+ */
+export interface CloudOperationsConnectionOptions {
+  /**
+   * The address of the Temporal Cloud Operations API to connect to, in `hostname:port` format.
+   *
+   * @default saas-api.tmprl.cloud:443
+   */
+  address?: string;
+
+  /**
+   * TLS configuration. TLS is required for connecting to the Temporal Cloud Operations API.
+   *
+   @default true
+   */
+  tls?: Pick<TLSConfig, 'serverNameOverride'> | true;
+
+  /**
+   * GRPC Channel arguments
+   *
+   * @see option descriptions {@link https://grpc.github.io/grpc/core/group__grpc__arg__keys.html | here}
+   *
+   * By default the SDK sets the following keepalive arguments:
+   *
+   * ```
+   * grpc.keepalive_permit_without_calls: 1
+   * grpc.keepalive_time_ms: 30_000
+   * grpc.keepalive_timeout_ms: 15_000
+   * ```
+   *
+   * To opt-out of keepalive, override these keys with `undefined`.
+   */
+  channelArgs?: grpc.ChannelOptions;
+
+  /**
+   * {@link https://grpc.github.io/grpc/node/module-src_client_interceptors.html | gRPC interceptors}
+   * which will be applied to every RPC call performed by this connection. By default, an interceptor
+   * will be included which automatically retries retryable errors. If you do not wish to perform
+   * automatic retries, set this to an empty list (or a list with your own interceptors). If you want
+   * to add your own interceptors while keeping the default retry behavior, add this to your list of
+   * interceptors: `makeGrpcRetryInterceptor(defaultGrpcRetryOptions())`.
+   *
+   * See:
+   * - {@link makeGrpcRetryInterceptor}
+   * - {@link defaultGrpcRetryOptions}
+   */
+  interceptors?: grpc.Interceptor[];
+
+  /**
+   * Optional mapping of gRPC metadata (HTTP headers) to send with each request to the server. An
+   * `Authorization` header set through `metadata` will be ignored and overriden by the value of the
+   * {@link apiKey} option.
+   *
+   * In order to dynamically set metadata, use {@link CloudOperationsConnection.withMetadata}
+   */
+  metadata?: Metadata;
+
+  /**
+   * API key for Temporal. This becomes the "Authorization" HTTP header with "Bearer " prepended.
+   *
+   * You may provide a static string or a callback. Also see {@link CloudOperationsConnection.withApiKey}
+   * or {@link Connection.setApiKey}
+   */
+  apiKey: string | (() => string);
```

Comment:
Interesting argument, but that's unfortunately non-trivial. We can consider that at a later point (note that this is really a copy-paste of `@temporalio/client`'s `Connection` class, so that's not really a new thing).